autonav_ws/src/autonav_odometry/src/particlefilter.py

```diff
@@ -0,0 +1,102 @@
+### particlefilter.py from last year ###
+# TODO could all the for-each-particle math in this code be done ✨🚀faster🚀✨ with numpy than vanilla python?
+from autonav_msgs.msg import MotorFeedback, GPSFeedback, Position
+import numpy as np
+import math
+import random
+
+
+class Particle:
+    def __init__(self, x=0, y=0, theta=0, weight=1) -> None:
+        self.x = x
+        self.y = y
+        self.theta = theta
+        self.weight = weight
+
+
+class ParticleFilter:
+    def __init__(self, latitudeLength, longitudeLength) -> None:
+        self.num_particles = 750
+        self.gps_noise = 0.45
+        self.odom_noise = (0, 0, 0.1)
+        # TODO should these parameters be in a config file in autonav_odometry/resource?
+        self.init_particles()
```

Comment:
bit of a redundant call (as you call onReset() in __init__ for FiltersNode line 34) but it's not that big of a deal.
what's a bigger deal is using the seed heading, which I think. . . actually, nevermind, because init_particles is called in FiltersNode after initialization when we would actually be able to send it a seed heading, so that's a problem for FiltersNode not ParticleFilter